Public/Src/Pips/Dll/Builders/ProcessBuilder.cs

```diff
@@ -122,6 +122,9 @@ public class ProcessBuilder : IDisposable
         /// <nodoc />
         public Options Options { get; set; }
 
+        /// <nodoc />
+        public ReadOnlyArray<AbsolutePath> PreserveOutputWhitelist { get; set; } = ReadOnlyArray<AbsolutePath>.Empty;

```

Comment:
>PreserveOutputWhitelist [](start = 43, length = 23)

Do we need some error checking?
1. PreserveOutputWhitelist cannot be set unless PreserveOutputs is enabled
2. Members of PreserveOutputsWhitelist must be actual declared outputs and not stray paths or typos

Reply:
This may be related to the fingerprint question that I wrote in some other comment.

Suppose that the whitelist affect weak fingerprint. Most likely We don't want any addition or removal of entries in the whitelist that have nothing to do with our outputs to affect our cacheability, right?

---
In reply to: [287428192](https://github.com/microsoft/BuildXL/pull/369#discussion_r287428192) [](ancestors = 287428192)

Reply:
it is now affecting WF and added an unit test for that. 

I also added error-checking. 

---
In reply to: [287442996](https://github.com/microsoft/BuildXL/pull/369#discussion_r287442996) [](ancestors = 287442996,287428192)